src/components/drawing/symbol_defs.js

```diff
@@ -10,544 +11,781 @@ var d3 = require('@plotly/d3');
  * add both and you get both
  */
 
+
+var emptyPath = 'M0,0Z';
+var sqrt2 = Math.sqrt(2);
+var sqrt3 = Math.sqrt(3);
+var PI = Math.PI;
+var cos = Math.cos;
+var sin = Math.sin;
+
 module.exports = {
     circle: {
         n: 0,
-        f: function(r) {
-            var rs = d3.round(r, 2);
-            return 'M' + rs + ',0A' + rs + ',' + rs + ' 0 1,1 0,-' + rs +
-                'A' + rs + ',' + rs + ' 0 0,1 ' + rs + ',0Z';
+        f: function(r, angle, standoff) {
+            if(skipAngle(angle)) return emptyPath;
+
+            var rs = round(r, 2);
+            var circle = 'M' + rs + ',0A' + rs + ',' + rs + ' 0 1,1 0,-' + rs + 'A' + rs + ',' + rs + ' 0 0,1 ' + rs + ',0Z';
+            return standoff ? align(angle, standoff, circle) : circle;
         }
     },
     square: {
         n: 1,
-        f: function(r) {
-            var rs = d3.round(r, 2);
-            return 'M' + rs + ',' + rs + 'H-' + rs + 'V-' + rs + 'H' + rs + 'Z';
+        f: function(r, angle, standoff) {
+            if(skipAngle(angle)) return emptyPath;
+
+            var rs = round(r, 2);
+            return align(angle, standoff, 'M' + rs + ',' + rs + 'H-' + rs + 'V-' + rs + 'H' + rs + 'Z');
```

Comment:
🌴 AFAICT all of these functions - with the exception of the circle - have the form:
```js
if(skipAngle(angle)) return emptyPath;
var basePath = /* something that only depends on r */
return align(angle, standoff, basePath);
```
So we could have kept the original functions and just wrapped them:
```js
f: addAngleAndStandoff(function(r) { ... })
```

That said, I'm worried about calling `parse-svg-path` and reconstructing the path for every marker being a big performance hit. Can't we keep the path unchanged but add a transform to the element?

Reply:
> That said, I'm worried about calling `parse-svg-path` and reconstructing the path for every marker being a big performance hit. Can't we keep the path unchanged but add a transform to the element?

Addressed in https://github.com/plotly/plotly.js/pull/6297/commits/da0234e87594cd5ce5aac1f034e2c56eea2596dc.

Reply:
> Can't we keep the path unchanged but add a transform to the element?

In fact on a separate branch I did try adding transforms before opening the PR. But that approach required additional work at each trace level (can of worms). It wasn't that successful in the end as I need to add style calls in plot api. I also had to skip animations. So in the end I decided to move with this alternative which also supports Web-GL traces.